### PR TITLE
Node consistency 51 60

### PR DESCRIPTION
**Pull request by martinstacey**
### Purpose
The following PR addresses nodes' names and descriptions to make them more descriptive for beginners.

**51) NODE: String.Replace**
-CHANGES: 
Change input and output from str to string

**52) NODE: String.Remove**
-CHANGES: 
Change input and output from str to string
-The count has no default value, so will eliminate description part where it specifies what happends if you dont have a value 

**53) NODE: String.Insert**
-CHANGES: 
Change input and output from str to string

**54) NODE: String.AllIndicesOf**
-CHANGES: 
Change input and output from str to string
Description: Finds a list of zero-based indexes of occurrence of a sub-string inside a string.
InputDescriptions:
string: A string to search in.
searchFor: Substring to search for.
ignoreCase: Whether or not comparison takes case into account.
Output Description:
indexList: List of indices or ints where substring is found
Search: all indices of,find substring,where,search

**55) NODE: String.Join**
-CHANGES: 
Change output from str to string

**56) NODE: String.Concat**
-CHANGES: 
Change output from str to string

**57) NODE: Python Script From String**
No changes needed

**58) NODE: Python Script**
No changes needed

**59) NODE: ScopeIf**
CHANGES:
Description: Returns the input of either the True or False input depending on what boolean value is toggled in the test input.
Input Description: 
Test: Boolean test
True: Returned if test is true
False: Returned if test is false

**60) NODE: LoopWhile**
CHANGES:
Inputs:
name initialValue
description: initial value for loop</param>
Outpus:
name: result
description: result from loop

**61) NODE: If (solved automatically by solving node 59)**
CHANGES:
Description: Returns the input of either the True or False input depending on what boolean value is toggled in the test input.
Input Description: 
Test: Boolean test
True: Returned if test is true
False: Returned if test is false

image

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers
@SHKnudsen
@StudioLE 
@MarkThorley 
@anrova 

### FYIs
-No FYI

### CODE REVIEW

File: src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx [hunk 1]
```diff
@@ -385,7 +385,7 @@
     <comment>Description for Or</comment>
   </data>
   <data name="PortDataFalseBlockToolTip" xml:space="preserve">
-    <value>False block</value>
+    <value>Block returned if test is false</value>
```
Comment (SHKnudsen) [inline src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx]: what do we mean by `Block`?

File: src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx [hunk 2]
```diff
@@ -397,13 +397,13 @@
     <value>operand</value>
   </data>
   <data name="PortDataResultToolTip" xml:space="preserve">
-    <value>result</value>
+    <value>Result block produced</value>
```
Comment (SHKnudsen) [inline src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx:400]: `block`?

File: src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx [hunk 3]
```diff
@@ -397,13 +397,13 @@
...
   </data>
   <data name="PortDataTestBlockToolTip" xml:space="preserve">
-    <value>Test block</value>
+    <value>Boolean to determine what code block is returned</value>
```
Comment (SHKnudsen) [inline src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx]: code block is confusing as it sounds like its returning a `CodeBlock`, is this for the `scopeIf` node?

File: src/Libraries/CoreNodes/String.cs [hunk 4]
```diff
@@ -310,15 +310,22 @@ public static int IndexOf(string str, string searchFor, bool ignoreCase = false)
...
-
-        public static int[] AllIndicesOf(string str, string searchFor, bool ignoreCase = false)
+        /// <summary>
+        /// Finds list of indices where sub-string appears inside a string.
+        /// </summary>
+        /// <param name="string">A string to search in.</param>
+        /// <param name="searchFor">Substring to search for.</param>
+        /// <param name="ignoreCase">Whether or not comparison takes case into account.</param>
+        /// <returns name="indexList">List of indices (integers) where substring is found</returns>
+        /// /// <search>all indices of,find substring,where,search</search>
```
Comment (SHKnudsen) [inline src/Libraries/CoreNodes/String.cs]: looks like there are some extra `///` here

Comment (martinstacey) [reply src/Libraries/CoreNodes/String.cs]: addressed in 
64fe05b

File: src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx (hunk 1 above)
Comment (martinstacey) [reply src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx]: addressed in 
1520bcc

File: src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx (hunk 2 above)
Comment (martinstacey) [reply src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx:400]: addressed in 
1520bcc

File: src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx (hunk 3 above)
Comment (martinstacey) [reply src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx]: addressed in 
1520bcc

**Pull request merged**